datasets/nlu_evaluation_data/nlu_evaluation_data.py

```diff
@@ -0,0 +1,234 @@
+                        "iot_wemo_off",
+                        "iot_wemo_on",
+                        "lists_createoradd",
+                        "lists_query",
+                        "lists_remove",
+                        "music_dislikeness",
+                        "music_likeness",
+                        "music_query",
+                        "music_settings",
+                        "news_query",
+                        "play_audiobook",
+                        "play_game",
+                        "play_music",
+                        "play_podcasts",
+                        "play_radio",
+                        "qa_currency",
+                        "qa_definition",
+                        "qa_factoid",
+                        "qa_maths",
+                        "qa_stock",
+                        "recommendation_events",
+                        "recommendation_locations",
+                        "recommendation_movies",
+                        "social_post",
+                        "social_query",
+                        "takeaway_order",
+                        "takeaway_query",
+                        "transport_query",
+                        "transport_taxi",
+                        "transport_ticket",
+                        "transport_traffic",
+                        "weather_query",
+                    ]
+                ),
+            }
+        )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+            post_processed=datasets.info.PostProcessedInfo(
+                features=post_process_features,
+            ),
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        train_path = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": train_path}),
+        ]
+
+    def _post_process(self, dataset, resources_paths):
+        # remove empty examples
+        dataset = dataset.filter(lambda ex: True if ex["answer_annotation"] != "null" else False, with_indices=False)
+        # remove annotations from examples and define label
+        dataset = dataset.map(
+            lambda ex: dict(
+                ex,
+                text=remove_annotations(ex["answer_annotation"]),
+                label=self.info.post_processed.features["label"].encode_example(
+                    define_intent_name(ex["scenario"], ex["intent"])
+                ),
+            ),
+            batched=False,
+            with_indices=False,
+            features=datasets.Features(self.info.features, **self.info.post_processed.features),
+        )
+        # drop unused columns
+        dataset = dataset.remove_columns(["answer_annotation", "intent"])
+        return dataset
```

Comment:
In general we do simple processing of examples directly in `_generate_examples`.
Could you do this parsing/processing in `_generate_examples` instead of using `_post_process` please ?

Reply:
@lhoestq it shouldn't be a problem except filtering (remove empty examples). If I do this in the for loop in `_generate_examples` (as I did previously) summary of the dataset will have an incorrect number of rows

Reply:
Thanks for the change !
It's ok to have less rows as soon as you mention in the dataset card that you removed empty rows.
Overall the idea is to have everything in _generate_examples

Reply:
Ok so I will push one more change for that

Reply:
@lhoestq some problems with filtering in for loop. I tried to put something like this in the for loop in `_generate_examples`, but this hack does not pass tests since it produces a different dataset size. Any recommendation on how to do this properly?
```
                if answer_annotation == "null":
                    continue

                yield id_, {
                    "text": remove_annotations(answer_annotation),
                    "scenario": scenario,
                    "label": define_intent_name(scenario, intent),
                }
```

Reply:
I think you can fix this by updating the number of expected examples in the dataset metadata in `dataset_infos.json`:
```
datasets-cli test ./datasets/nlu_evaluation_data --save_infos --all_configs --ignore_verifications
```

Reply:
thanks for a quick reply!